src/main.js

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env node
+
+let debugModule = require('debug');
+let debug = debugModule('aws-provisioner:main');
+let aws = require('aws-sdk-promise');
+let _ = require('lodash');
+let path = require('path');
+
+let taskcluster = require('taskcluster-client');
+let base = require('taskcluster-base');
+
+let workerType = require('./worker-type');
+let secret = require('./secret');
+let workerState = require('./worker-state');
+let AwsManager = require('./aws-manager');
+let provision = require('./provision');
+let exchanges = require('./exchanges');
+let v1 = require('./api-v1');
+let series = require('./influx-series');
+
+process.on('unhandledRejection', err => {
+  debug('[alert-operator] UNHANDLED REJECTION!\n' + err.stack || err);
+});
+
+let load = base.loader({
+  cfg: {
+    requires: ['profile'],
+    setup: ({profile}) => base.config(profile),
+  },
+
+  WorkerType: {
+    requires: ['cfg'],
+    setup: async ({cfg}) => {
+      let WorkerType = workerType.setup({
+        table: cfg.app.workerTypeTableName,
+        credentials: cfg.azure,
+        context: {
+          keyPrefix: cfg.app.awsKeyPrefix,
+          provisionerId: cfg.app.id,
+          provisionerBaseUrl: cfg.server.publicUrl + '/v1',
+          pubKey: cfg.app.awsInstancePubkey,
+        },
+      });
+      await WorkerType.ensureTable();
+      return WorkerType;
+    },
+  },
+
+  WorkerState: {
+    requires: ['cfg'],
+    setup: async ({cfg}) => {
+      let WorkerState = workerState.setup({
+        table: cfg.app.workerStateTableName,
+        credentials: cfg.azure,
+      });
+      await WorkerState.ensureTable();
+      return WorkerState;
+    },
+  },
+
+  Secret: {
+    requires: ['cfg'],
+    setup: async ({cfg}) => {
+      let Secret = secret.setup({
+        table: cfg.app.secretTableName,
+        credentials: cfg.azure,
+      });
+      await Secret.ensureTable();
+      return Secret;
+    },
+  },
+
+  validator: {
+    requires: ['cfg'],
+    setup: async ({cfg}) => {
+      return await base.validator({
+        folder: path.join(__dirname, '..', 'schemas'),
+        constants: require('../schemas/constants'),
+        publish: cfg.app.publishMetaData === 'true',
```

Comment:
newest validator guess this using `NODE_ENV == 'production'`... I think...

@imbstack, can you help review this main.js, specifically, you are 100% sure about what is best practice and not...